rcl/src/rcl/arguments.c

```diff
@@ -639,6 +679,20 @@ rcl_parse_arguments(
     }
   }
 
+  // Shrink logger settings of log levels
```

Comment:
Please create a function for this:

```c
rcl_ret_t rcl_log_levels_shrink_to_size(const char * name, rcl_log_severity_t level);
```

In that way, things are much easier to unit test.